pkg/commands/workload_create.go

```diff
@@ -68,9 +69,21 @@ func (opts *WorkloadCreateOptions) Exec(ctx context.Context, c *cli.Config) erro
 		workload.Namespace = opts.Namespace
 	}
 
-	if err := c.Get(ctx, client.ObjectKey{Namespace: workload.Namespace, Name: workload.Name}, &cartov1alpha1.Workload{}); !apierrs.IsNotFound(err) {
-		c.Printf("%s workload %q already exists\n", printer.Serrorf("Error:"), fmt.Sprintf("%s/%s", workload.Namespace, workload.Name))
-		return cli.SilenceError(err)
+	existingWorkload := &cartov1alpha1.Workload{}
+
+	if err := c.Get(ctx, client.ObjectKey{Namespace: workload.Namespace, Name: workload.Name}, existingWorkload); err != nil {
+		// return err, except when not found
+		if !apierrs.IsNotFound(err) {
+			return err
+		}
+	}
+
+	// check if the workload exists
+	if existingWorkload != nil {
```

Comment:
you can merge both if conditions

Reply:
If you can merge it, we avoid an untested branch. If existingWorkload is nil, then rest of the conditions are not executed either in Go.

Reply:
@shashwathi Are you asking to merge `err` and `existingWorkload` checks in a single `if` statement? These are two different objects and do not follow your comment.

Reply:
```suggestion
	if existingWorkload != nil && existingWorkload.Name == workload.Name && existingWorkload.Namespace == workload.Namespace {
	c.Printf("%s workload %q already exists\n", printer.Serrorf("Error:"), fmt.Sprintf("%s/%s", workload.Namespace, workload.Name))
			return cli.SilenceError(err)
			}
```

Reply:
Both suggested and implemented change work in this particular `if` statement. However, the implemented code is easier to read. Therefore, we are going with the PR implementation. Thank you for the suggestion.